packages/components/src/internal/test.ts

```diff
@@ -44,3 +44,53 @@ export async function clickOnElement(
 
   await sendMouse({ type: 'click', position: [clickX, clickY] });
 }
+
+function determineMousePosition(el: Element, position: string, offsetX: number, offsetY: number) {
+  const { x, y, width, height } = el.getBoundingClientRect();
+  const centerX = Math.floor(x + window.pageXOffset + width / 2);
+  const centerY = Math.floor(y + window.pageYOffset + height / 2);
+  let clickX: number;
+  let clickY: number;
+
+  switch (position) {
+    case 'top':
+      clickX = centerX;
+      clickY = y;
+      break;
+    case 'right':
+      clickX = x + width - 1;
+      clickY = centerY;
+      break;
+    case 'bottom':
+      clickX = centerX;
+      clickY = y + height - 1;
+      break;
+    case 'left':
+      clickX = x;
+      clickY = centerY;
+      break;
+    default:
+      clickX = centerX;
+      clickY = centerY;
+  }
+
+  clickX += offsetX;
+  clickY += offsetY;
+  return { clickX, clickY };
+}
```

Comment:
Switch case statements are always hard to read in my opinien. How about a more simpler way like:

```
function determineMousePosition(el: Element, position: string, offsetX: number, offsetY: number) {
  const { x, y, width, height } = el.getBoundingClientRect();
  const centerX = Math.floor(x + window.pageXOffset + width / 2);
  const centerY = Math.floor(y + window.pageYOffset + height / 2);

  const positionMap = {
    top: { clickX: centerX, clickY: y },
    right: { clickX: x + width - 1, clickY: centerY },
    bottom: { clickX: centerX, clickY: y + height - 1 },
    left: { clickX: x, clickY: centerY },
  };

  const { clickX, clickY } = positionMap[position] || { clickX: centerX, clickY: centerY };

  return { clickX: clickX + offsetX, clickY: clickY + offsetY };
}
```

I didn't test my code, so there might be a need for further adjustments.

Reply:
@yoezlem 
Hm, I get that switches are not always the best to go. But for this particular situation I think the switch makes it easier to understand all the individual calculations compared to the map since they're more organised. What do you think?

Reply:
@karlbaumhauer @mariohamann what are your thoughts about that?

Reply:
Hm. I don't know. In a component I think I would go more with Özlem's solution but as this is apparently used only for tests I don't have strong feelings on it.

I would just suggest to move the function into `moveMouseOnElement` that it is more clear, that this is only used for this function.

Reply:
@mariohamann Actually, `determineMousePosition`'s logic is also used in `clickOnElement`. 

So I propose we keep the function and remove the duplicate code in `clickOnElement`. Now we have two functions using `determineMousePosition`